### PR TITLE
chaincfg: Introduce new type DNSSeed 

### PR DESCRIPTION
**Pull request by dnldd**
This PR contains the following upstream commits:

- [bca9877](https://github.com/decred/dcrd/commit/bca9877796e9590adca69818e3049fbcff9a08af)
  - NOOP
- [a09d052](https://github.com/decred/dcrd/commit/a09d052f96f5dc19d014e2ee8243ca87520882b8)
  - NOOP
- [0d508e6](https://github.com/decred/dcrd/commit/0d508e652258c10230c22651de1ce2c395f695d4)
  - NOOP
- [f161d6b](https://github.com/decred/dcrd/commit/f161d6b69e9adb97a04e51c28f26eb614f9f01a2)

---
DNSSeed defines a DNS Seed with a hostname and whether it supports
filtering by service flag bits.

### CODE REVIEW

**Comment by dajohi** [comment]
@dnldd did you run this?

**Comment by dnldd** [comment]
@dajohi yes, run it on testnet. dcrd syncedjust fine.

**Comment by dajohi** [comment]
sorry, no.  just wanted to make sure you ran it with peers.json removed.

File: chaincfg/params.go [hunk 1]
```diff
@@ -1086,7 +1096,12 @@ var (
 	hdPrivToPubKeyIDs = make(map[[4]byte][]byte)
 )
 
-// Register registers the network parameters for a Decred network.  This may
+// String returns the hostname of the DNS seed in human-readable form.
+func (d DNSSeed) String() string {
+	return d.Host
+}
+
+// Register registers the network parameters for a Bitcoin network.  This may
```
Comment (davecgh) [inline chaincfg/params.go]: Looks like you accidentally reverted the Decred to Bitcoin.  s/Bitcoin/Decred/

**Pull request merged**